broker/src/main/java/io/camunda/zeebe/broker/bootstrap/JobStreamServiceStep.java

```diff
@@ -34,34 +37,23 @@ void startupInternal(
       final BrokerStartupContext brokerStartupContext,
       final ConcurrencyControl concurrencyControl,
       final ActorFuture<BrokerStartupContext> startupFuture) {
-    final var clusterServices = brokerStartupContext.getClusterServices();
-    final var scheduler = brokerStartupContext.getActorSchedulingService();
-
-    final StreamRegistry<JobActivationProperties> registry = new StreamRegistry<>();
-    final var requestHandler = new StreamApiHandler<>(registry, DummyActivationProperties::new);
-    final var server =
-        new JobStreamApiServer(clusterServices.getCommunicationService(), requestHandler);
-    final var streamer =
-        new JobGatewayStreamer(
-            clusterServices.getEventService(), clusterServices.getCommunicationService(), registry);
-    final var service = new JobStreamService(server, streamer);
-
-    // only register listener and apply service to context if the actors are scheduled successfully
-    final var result =
-        Stream.of(scheduler.submitActor(server), scheduler.submitActor(streamer))
-            .collect(new ActorFutureCollector<>(concurrencyControl));
+    final RemoteStreamService<JobActivationProperties, ActivatedJob> remoteStreamService =
+        new TransportFactory(brokerStartupContext.getActorSchedulingService())
+            .createRemoteStreamServer(
+                brokerStartupContext.getClusterServices().getCommunicationService(),
+                DummyActivationProperties::new,
+                brokerStartupContext.getClusterServices().getEventService());
+
+    final var startFuture = remoteStreamService.start();
     concurrencyControl.runOnCompletion(
-        result,
-        (ok, error) -> {
-          if (error != null) {
-            startupFuture.completeExceptionally(error);
-            return;
-          }
-
-          clusterServices.getMembershipService().addListener(server);
-          brokerStartupContext.setJobStreamService(service);
-          startupFuture.complete(brokerStartupContext);
+        startFuture,
+        (streamer, error) -> {
+          // TODO: Put remoteStreamService into context
```

Comment:
:wrench: If we're always only going to use it as the job stream service, then we should put the wrapper (either the JobGatewayStreamer or something) directly here, and just have it manage the lifecycle of the stream service